packages/components/src/DataList/DataList.tsx

```diff
@@ -53,8 +53,7 @@ export function DataList<T extends DataListObject>({
       </div>
       {layout &&
         elementData.map((child, i) => (
-          // TODO: Don't use index as key. Might have to force an ID on the data JOB-76773
-          <div className={styles.listItem} key={i}>
+          <div className={styles.listItem} key={data[i].id}>
```

Comment:
this feels weird to use the index to lookup the id rather than using `child.id` when we are already iterating over the data set

I see in `generateListItemElements` we're modifying this returned array to be something else entirely, so I get why it's necessary with the way this is set up... buuut that is something we control

one thing we could do is instead change the data structure returned from `generateListItemElements` to have 2 keys: `id` and `elements` or something like that. then we'd be able to do `child.id` and then `layout(child.elements)`

that said my argument is mostly "this feels weird". I haven't tested this out with a massive list to see if there are any actual performance concerns with an extra lookup on a big data set

Reply:
I honestly, just didn't want to fight with types on this one haha

Reply:
fair.

ok after doing what I hope is a reasonable experiment at seeing the performance, to me it looks negligible to lookup the id again rather than using an id that we have provided

<img width="1022" alt="image" src="https://github.com/GetJobber/atlantis/assets/11843215/8f3865ba-227d-48f8-ab73-972b10c5fa2f">

Reply:
oooh fancy